### PR TITLE
[docs] Mounting config file in container

### PR DESCRIPTION
**Pull request by daenney**
# Description

This adds a small section clarifying how to mount a config.yaml in a container.

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [ ] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [ ] I/we have performed a self-review of added code.
- [ ] I/we have written code that is legible and maintainable by others.
- [ ] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [ ] I/we have added tests that cover new code.
- [ ] I/we have run tests and they pass locally with the changes.
- [ ] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

**Comment by daenney** [comment]
https://gotosocial--2622.org.readthedocs.build/en/2622/configuration/#mounting-in-container

File: docs/configuration/index.md [hunk 1]
```diff
@@ -22,6 +22,38 @@ An example configuration file, with an explanation of each of the config fields,
...
+#### Mounting in container
+
+It can be necessary to have a `config.yaml` in a container, as some settings aren't easily managed with environment variables or command line flags.
+
+To do so, create the `config.yaml` on host, and then mount it anywhere you please inside the container. Then, set the `command` to pass the location of the configuration file. For `docker-compose.yaml`, you can amend it like so:
+
+```yaml
+services:
+  gotosocial:
+    command: ["--config-path", "/gotosocial/config.yaml"]
```
Comment (tsmethurst) [inline docs/configuration/index.md:34]: Might be worth mentioning that it's also possible to set the GTS_CONFIG_PATH env variable pointing to the path

Comment (daenney) [reply docs/configuration/index.md:34]: Ah yeah, good point

Comment (daenney) [reply docs/configuration/index.md:36]: Done!

**Pull request merged**